test/solc/CommandLineInterface.cpp

```diff
@@ -278,6 +278,113 @@ BOOST_AUTO_TEST_CASE(cli_not_a_file)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(linker)
+{
+	frontend::FileReader fileReader({},{},{});
+	fileReader.setSourceUnits({
+		{"bytecodeInput.bin", "565b6040516100579190610158565b60405180910390f35b600073__$b3157e226f2c4dddae135e6cab4ed4e747$__63b3de648b8360405"}
+	});
+
+	string hashString = "1234567890123456789012345678901234567890";
+	map<string, util::FixedHash<20>> libraries = {
+		{"test/libsolidity/semanticTests/libraries/stub.sol:L", FixedHash<20>("0x" + hashString)}
+	};
+
+	bool hasError = false;
+
+	CommandLineInterface::link(fileReader, libraries, hasError, cerr);
+	BOOST_TEST(!hasError);
+	BOOST_TEST(fileReader.sourceUnits().find("bytecodeInput.bin")->second.find(hashString) != string::npos);
```

Comment:
Why not just compare the result directly? Less chance of false positives by mistake and it's not that long anyway:

```suggestion
	map<string, string> expectedSources = {{
		"bytecodeInput.bin",
		"565b6040516100579190610158565b60405180910390f35b600073" + hashString + "63b3de648b8360405"
	}};
	BOOST_TEST(fileReader.sourceUnits() == expectedSources);
```